engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -446,6 +505,9 @@ namespace dmGameSystem
                 if (info->m_Version != component.m_BufferVersion)
                 {
                     info->m_Version = component.m_BufferVersion;
+                    // process position stream in buffer to find bounding box
+                    uint64_t stream_name = (component.m_Resource->m_PositionStreamId != 0) ? component.m_Resource->m_PositionStreamId : dmHashString64("position"); // use stream name (hash) from resource and fallback to "position" if this is not available
+                    UpdateMeshBounds(br->m_Buffer, stream_name);
```

Comment:
Yes, they will be. 

But dmBuffer()::SetBounds() is supposed to be a private method right ? User cannot call it. For now. 

Later, when we will make the api public we can:

1. Settle `dmBuffer::SetBounds(min,max)` api to `SetBounds(min,max, force=false)`. That is, add one more parameter for flexibility.
2. Add a version field (say buffer.m_BoundsVersion) in the buffer meta data. It will keep buffer version at the time SetBounds() was called. SetBounds() will check if the bound info it has is up to date using the version identifier and will not update the bounds if it is.
2. The component logic will stay the same. Call SetBounds() as it does currently. `force` will be false.
3. When the user wants to explicitly set the bounds, he/she calls SetBounds() with `force` set to true. 

@JCash

Reply:
I think the #2 is the only viable option.
The "force" flag doesn't really work I'd say.

I recommend you do a test in C++.

Reply:
I didn't have a good example handy, but this looks like it might work:
https://forum.defold.com/uploads/default/original/3X/f/b/fbb9c4abd3ed239f6066f912147c50744bdafa90.zip

Very simple test, creating a quad.
Test by setting bounds on the buffer directly, e.g. exaggerate the bounds, and then check that it's intact when arriving at the culling step.

Reply:
Thanks.

In line with our discussion i've updated the PR with the following:

1. New m_BoundingVersion field in `Buffer`. It gets m_ContentVersion's value when SetBounds() is called.
2. Made Get/SetBounds() public i.e. moved them to dmsdk/dlib/buffer.h
3. Introduced (private) dmBuffer::BoundsUpToDate(). It tells if the min/max bounds in Buffer are up-to-date by comparing the two version fields. I'll explain why we need this in a while.
4. comp_mesh.cpp::UpdateSetBounds() will only calculate and update buffer bounds if they are out of date.

#### Why BoundsUpToDate()

Even with the newly added Buffer.m_BoundingVersion it's not possible to avoid the automatic re-calculation of bounds triggered by the component. Suppose the user updates the buffer from C++, calls dmBuffer::UpdateContentVersion() and dmBuffer::SetBounds() accordingly. The mesh component won't know. It will detect the content version change in the buffer and (a) calculate new bounding box on its own, (b) try to update the buffer with this bounding box. We could avoid (b) if SetBounds() itself checks if there is a version mismatch inside the buffer. We can't avoid (a) though. 

That's the reason i added BoundsUpToDate(). Give some insight to the component so it won't mess with the bounds explicitly given by the user. I put it into dlib/buffer.h. That's not pubic API right ?